database/engine/metadata_log/metalogpluginsd.c

```diff
@@ -4,6 +4,48 @@
 #include "metadatalog.h"
 #include "metalogpluginsd.h"
 
+PARSER_RC metalog_pluginsd_host_action(
+    void *user, char *machine_guid, char *hostname, char *registry_hostname, int update_every, char *os, char *timezone,
+    char *tags)
+{
+    UNUSED(user);
+
+    if (strcmp(machine_guid, registry_get_this_machine_guid()) == 0)
+        return PARSER_RC_OK;
+
+    RRDHOST *host = rrdhost_find_by_guid(machine_guid, 0);
+    if (host)
+        return PARSER_RC_OK;
+
```

Comment:
We need code similar to `metalog_pluginsd_chart_action()` and `metalog_pluginsd_dimension_action()` like this:
```
    if (host) { /* It's a valid object */
        struct metalog_record record;
        struct metadata_logfile *metalogfile = state->metalogfile;

        uuid_copy(record.uuid, state->uuid);
        mlf_record_insert(metalogfile, &record);
        uuid_clear(state->uuid); /* Consume UUID */
    }

```

This is needed so that during log replay, the necessary `struct metalog_record` is created so that afterwards compaction won't miss it.

Reply:
You are right. At the comment there is extra check during compaction so that if a chart compaction is attempted without the host being compacted -- it is forced -- I will add it